scissors/weapon_selection.c

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file       weapon_selection.c
+ * @brief      weapon_selection
+ * 
+ * @author     Leonid Chebenko
+ * @version    1.0
+ */
+
+#include<stdio.h>
+#include<stdlib.h>
+#include<unistd.h>
+#include "weapon_selection.h"
+#include "game_logic.h"
+
+void weapon_selection()
+{
+        int player;
```

Comment:
The `extern` declaration of global variable `player` which defined in the `main.c` is not visible here so that compiler does not even warn about variable hiding.
You store the choice in this local variable but the `game_logic()` function uses global one.
So, initial `0` value is always used as a user choice.